phoenix-core/src/it/java/org/apache/phoenix/end2end/ArrayToJsonFunctionIT.java

```diff
@@ -0,0 +1,150 @@
+                            + "(pk, BOOLEAN_ARRAY, BYTE_ARRAY, DOUBLE_ARRAY, FLOAT_ARRAY, INT_ARRAY, LONG_ARRAY, SHORT_ARRAY, STRING_ARRAY)\n"
+                            + "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)");
+
+            stmt.setString(1, "valueOne");
+
+            // boolean array
+            Array boolArray = conn.createArrayOf("BOOLEAN", new Boolean[] { true,false });
+            int boolIndex = 2;
+            stmt.setArray(boolIndex, boolArray);
+            // byte array
+            Array byteArray = conn.createArrayOf("TINYINT", new Byte[] { 11, 22 });
+            int byteIndex = 3;
+            stmt.setArray(byteIndex, byteArray);
+            // double array
+            Array doubleArray = conn.createArrayOf("DOUBLE", new Double[] { 67.78, 78.89 });
+            int doubleIndex = 4;
+            stmt.setArray(doubleIndex, doubleArray);
+            // float array
+            Array floatArray = conn.createArrayOf("FLOAT", new Float[] { 12.23f, 45.56f });
+            int floatIndex = 5;
+            stmt.setArray(floatIndex, floatArray);
+            // int array
+            Array intArray = conn.createArrayOf("INTEGER", new Integer[] { 5555, 6666 });
+            int intIndex = 6;
+            stmt.setArray(intIndex, intArray);
+            // long array
+            Array longArray = conn.createArrayOf("BIGINT", new Long[] { 7777777L, 8888888L });
+            int longIndex = 7;
+            stmt.setArray(longIndex, longArray);
+            // short array
+            Array shortArray = conn.createArrayOf("SMALLINT", new Short[] { 333, 444 });
+            int shortIndex = 8;
+            stmt.setArray(shortIndex, shortArray);
+            // create character array
+            Array stringArray = conn.createArrayOf("VARCHAR", new String[] { "a", "b" });
+            int stringIndex = 9;
+            stmt.setArray(stringIndex, stringArray);
+            stmt.execute();
+            conn.commit();
+
+            stmt =
+                    conn.prepareStatement("SELECT pk, " +
+                            "array_to_json(boolean_array), " +
+                            "array_to_json(byte_array), " +
+                            "array_to_json(double_array), " +
+                            "array_to_json(float_array), " +
+                            "array_to_json(int_array), " +
+                            "array_to_json(long_array), " +
+                            "array_to_json(short_array)," +
+                            "array_to_json(string_array) FROM "
+                            + TABLE_WITH_ALL_ARRAY_TYPES);
+
+            ResultSet rs = stmt.executeQuery();
+            assertTrue(rs.next());
+
+            assertEquals("valueOne", rs.getString(1));
+            assertArrayToJson(rs, boolIndex, "[true,false]");
+            assertArrayToJson(rs, byteIndex, "[11,22]");
+            assertArrayToJson(rs, doubleIndex, "[67.78,78.89]");
+            assertArrayToJson(rs, floatIndex, "[12.23,45.56]");
```

Comment:
can you please check for the null value in the array. I am not sure whether we allow null value in Phoenix array or not.